runtime/js/40_write_file.js

```
@@ -33,15 +45,35 @@
       abortHandler = () => core.tryClose(cancelRid);
       options.signal[abortSignal.add](abortHandler);
     }
-    try {
-      await core.opAsync("op_write_file_async", {
-        path: pathFromURL(path),
-        data,
+    const path = pathFromURL(pathOrURL);
+    let fsFile = openFdCache.get(path);
+    const append = options.append ?? false;
```

`options.append === true`

---

Though, these would change observable behaviour, presumably changing throw errors into silent defaults.